test/gtest/ucp/test_ucp_request.cc

```diff
@@ -100,3 +102,392 @@ UCS_TEST_P(test_ucp_request, test_request_query)
 }
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_request, all, "all")
+
+class test_proto_reset : public ucp_test {
+public:
+    enum operation_e {
+        TAG,
+        RMA_GET,
+        RMA_PUT,
+        STREAM,
+        AM
+    };
+
+    test_proto_reset() : m_msg_size(UCS_KBYTE * 64), m_completed(0)
+    {
+        m_pending.head  = NULL;
+        m_pending.ptail = NULL;
+    }
+
+    void init() override
+    {
+        if (!m_ucp_config->ctx.proto_enable) {
+            UCS_TEST_SKIP_R("reset is not supported for proto v1");
+        }
+
+        ucs::scoped_setenv snd_buf("UCX_TCP_SNDBUF", "8K");
+        ucs::scoped_setenv ib_queue_len("UCX_IB_TX_QUEUE_LEN", "65");
```

Comment:
use modify_config

Reply:
does not work for UCT config

Reply:
are you sure? it should

Reply:
Invalid UCS configuration for TCP_SNDBUF : 8K, error message: No such element(-12)

Reply:
See here:
https://github.com/openucx/ucx/blob/4d8d7517c054e6f4274455105ee51d2a501e75fd/test/gtest/ucp/test_ucp_am.cc#L1453